addon/experiment-api/contactsApi.js

```diff
@@ -53,6 +100,61 @@ var convContacts = class extends ExtensionCommon.ExtensionAPI {
             args
           );
         },
+        async composeNew(properties) {
+          const window =
+            properties.windowId !== null
+              ? windowManager.get(properties.windowId, context).window
+              : Services.wm.getMostRecentWindow("mail:3pane");
+          const { to } = properties;
+          composeMessageTo(to, window.gFolderDisplay.displayedFolder);
+        },
+        async showMessagesInvolving(options) {
+          const window =
+            options.windowId !== null
+              ? windowManager.get(options.windowId, context).window
+              : Services.wm.getMostRecentWindow("mail:3pane");
+
+          const { name, email } = options;
+
+          const collection = await glodaQuery({
+            kind: "email",
+            value: email,
+            query: "NOUN_IDENTITY",
+          });
+          if (!collection.items.length) {
+            return;
+          }
+
+          const result = await glodaQuery({
+            query: "NOUN_MESSAGE",
+            involvesItems: collection.items,
+          });
+
+          const browser = BrowserSim.getBrowser();
+          let tabmail = window.document.getElementById("tabmail");
+          tabmail.openTab("glodaList", {
+            collection: result,
+            title: browser.i18n.getMessage("involvingTabTitle", [name]),
+            background: false,
+          });
+        },
+        // For some reason, this needs to be implemented as an async function
```

Comment:
Side note: We don't have to do these as async, but the general design philosophy for the APIs is to have them as async, then if they do need changing later, extensions don't have to adopt.

Reply:
I put that comment there because I originally defined it as synchronous, but it didn't work. I couldn't figure out why (I wasn't getting a return value or something). I turned it to async and it worked. Still a mystery to me.

Reply:
Ah, well generally we do want these async (even if they're only sync), but the reason it has to be is because of the API definition in the schema, there's an async flag. We also want them async as eventually they'll probably be going across process, so async will be more efficient overall.

Reply:
I had set `"async": false` in the schema though...In any case. I'm glad it's correct as it is. Should I remove the comment?

Reply:
Ah yes, you could do.